ruby_event_store/lib/ruby_event_store/event.rb

```diff
@@ -6,6 +6,8 @@ def initialize(event_id: SecureRandom.uuid, metadata: nil, data: nil)
       @event_id = event_id.to_s
       @metadata = Metadata.new(metadata.to_h)
       @data     = data.to_h
+      self.correlation_id ||= @event_id
```

Comment:
Is this for subclassing of Event and having set corelation_id and causation_id before super() call ?

Reply:
Makes sens, I was thinking that this implies order of using super() in subclass initializer, this probably doesn't matter that much.

Reply:
`I want every message to always have these 3 IDs so I never need to think that they could be nil and workaround for that on any higher layer.` I have no idea what's the proper solution yet, but it sounds like there can be something which avoids this problem.

It doesn't make sense that we set the causation_id or correlation_id to the event itself.

Reply:
> It doesn't make sense that we set the causation_id or correlation_id to the event itself.

Thanks! Obviously it made sense to at least someone (me)...